simulator/src/main/java/simblock/simulator/Simulator.java

```diff
@@ -46,6 +46,18 @@ public class Simulator {
 
   private static AbstractConsensusAlgo consensusAlgo = null;
 
+  private static int nextNodeIdx = 0;
+
+  public static Node getNextNode(){
+    // the size of readyNodes will changes as new nodes receive blocks and are marked ready
+    // if we just call .get(nextNodeIdx % size) as opposed to explicitly resetting the value, we could see unexpected behavior
+    if (nextNodeIdx >= readyNodes.size()){
```

Comment:
Instead of adding a new array, add a field to the `Node` class, that indicates whether the node is ready. 
After you choose the id of the next node, if the node is active then you can nominate it as miner, otherwise, you need to loop through the next elements of the array until you find one that is ready.